addons/pos_self_order/models/pos_session.py

```diff
@@ -33,3 +33,21 @@ def _loader_params_product_product(self):
         res = super()._loader_params_product_product()
         res['search_params']['fields'].append('self_order_available')
         return res
+
+    def _pos_data_process(self, loaded_data):
+        """
+        This is where we need to process the data if we can't do it in the loader/getter
+        """
+        super()._pos_data_process(loaded_data)
+        loaded_data["company_has_kiosk"] = (
```

Comment:
- rename 'company_has_kiosk' to 'company_has_self_ordering'.
- update the search below to include 'mobile mode'.